File: JAVA/Data Structure/Queue.java
import java.util.Scanner;

class Node
{
    int data ;
    Node next ;

    public Node(int iNo)
    {
        data = iNo ;
        next = null ;
    }
}

class Queuee 
{
    // characterstics
    public Node head ;
    public int iCountNode ;

    // constructor
    public Queuee()
    {
        head = null ;
        iCountNode = 0 ;
    }

    // countNode method is used for count the peresent elements in Queue
    public int countNode()
    {
        if(head == null)
        {
            return 0 ;
        }

        return iCountNode ;
    }

    // displayQueue method is used for display the elements which are present inside the Queue
    public void displayQueue()
    {
        if(head == null)
        {
            System.out.println("NULL : ! Queue is empty !");
            return ;
        }

        Node temp = null ;

        temp = head ;
 
        System.out.println("------ Queue (Front to Rear) ------");
        while(temp != null)
        {
            System.out.print("| " + temp.data + " |<-");
            temp = temp.next ;
        }
        System.out.println("null");
    }

    // enQueue method is used for insert the new node inside the Queue at Last position
    public void enQueue(int iNo)
    {
        Node newn = null ;

        newn = new Node(iNo) ;

        if(head == null)
        {
            head = newn ;
        }
        else
        {
            Node temp = null ;

            temp = head ;

            while(temp.next != null)
            {
                temp = temp.next ;
            }
            temp.next = newn ;
        }
        iCountNode++ ;

        System.out.println("Data inserted succesfully.");
    }

    // deQueue method is used to Delete the first node of the Queue
    public int deQueue()
    {
        int no = -1 ;

        if(head == null)
        {
            System.out.println("Unable to delete because Queue is already empty !!");
            return no ;
        }

        if(head.next == null)
        {
            no = head.data ;

            head = null ;
        }
        else
        {
            no = head.data ;

            head = head.next ;
        }
        iCountNode-- ;

        System.out.println("Data deleted succesfully.");

        return no ;
    }

    // DeleteAllNodes method is used to delete all nodes and make Queue empty
    public void emptyQueue()
    {
        if(head == null)
        {
            System.out.println("Unable to delete because   Queue is empty !!");
            return ;
        }

        Node temp = head ;
        Node nextnode = null ;

        while(temp != null)
        {
            nextnode = temp.next ;
            temp.next = null ;

            temp = nextnode ;
        }

        head = null ;
        iCountNode = 0 ;

        System.out.println("All Data deleted succesfully, Now Queue is empty");
    }
}

class Queue
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        int iRet = 0 , iOption = 0 , iData = 0 , iDeleteData = 0 ;

        Queuee qobj = new Queuee();

        while(true)
        {
            System.out.println("-----------------------------------------------------------------");
            System.out.println("0 : Terminate the Application");
            System.out.println("1 : Display the Queue ");
            System.out.println("2 : Count the total nodes of Queue");
            System.out.println("3 : EnQueue the data (insert the data)");
            System.out.println("4 : Dequeue the data (delete the data)");
            System.out.println("5 : Delete the all data from Queue (OR) make Queue empty");
            System.out.println("-----------------------------------------------------------------");
            System.out.println("Enter the option : ");
            iOption = sobj.nextInt();
            System.out.println("-----------------------------------------------------------------");

            switch(iOption)
            {
                case 0 :
                    System.out.println("Module is closing");
                    sobj.close();
                    return ;

                case 1 :
                    qobj.displayQueue();
                    break ;
                
                case 2 :
                    iRet = qobj.countNode(); 
                    System.out.println("Total nodes in queue are : "+iRet);
                    break ;

                case 3 :
                    System.out.println("Enter the data : ");
                    iData = sobj.nextInt();

                    qobj.enQueue(iData);
                    break ;

                case 4 :
                    iData = qobj.deQueue();
                    System.out.println("\n deQueued element is : "+ iData + "\n");
                    break ;

                case 5 :
                    qobj.emptyQueue();
                    break ;

                default :
                    System.out.println("!! Invalid Option !!");
                    break ;
            }     
        }
    }
}

File: C++/Data Structure/Queue.cpp
#include<iostream>
using namespace std ;

struct node
{
    int data ;
    struct node* next ;
};

typedef struct node NODE ;
typedef struct node* PNODE ;

class Queuee
{
    public:
        int iCountNode ;
        PNODE head ;

        Queuee()
        {
            iCountNode = 0 ;
            head = NULL ;
        }

        int CountNode()
        {
            return iCountNode ;
        }

        void Display()
        {
            PNODE temp = NULL ;

            temp = head ;

            while(temp != NULL)
            {
                cout<<"| "<<temp->data<<" |<-";
                temp = temp->next ;
            }
            cout<<"NULL\n";
        }

        void EnQueue(int iNo)
        {
            PNODE newn = NULL ;

            newn = new NODE ;

            newn->data = iNo ;
            newn->next = NULL ;

            if(head == NULL)
            {
                head = newn ;
            }
            else
            {
                PNODE temp = NULL ;

                temp = head ;

                while(temp->next != NULL)
                {
                    temp = temp->next ;
                }
                temp->next = newn ;
            }
            iCountNode++ ;
        }

        int DeQueue()
        {
            PNODE target = NULL ; 
            int no = -1 ;

            if(head == NULL)
            {
                cout<<"Queue is empty , Unable to Delete element !!";
                return -1 ;
            }
            else if(head->next == NULL)
            {
                no = head->data ;

                target = head ;

                delete target;
                head = NULL ;
            }
            else
            {
                no = head->data ;

                target = head ;
                head = head->next ;

                delete target ;
            }

            iCountNode-- ;

            return no ;
        }
};

int main()
{
    int iOption = 0 , iRetCountNode = 0 , iNo = 0 , iRetDeQueue = 0 ;

    Queuee qobj ;

    cout<<"<-- You are using Queue module -->\n";
    cout<<"--------------------------------------------------------\n";

    while(1)
    {
        cout<<"______________________________________________________\n";
        cout<<"1 Display the Queue\n";
        cout<<"2 Count the total elements from the Queue\n";
        cout<<"3 EnQueue(Insert at first position) the element into the Queue\n";
        cout<<"4 DeQueue(delete from first position) the element from thr Queue\n";
        cout<<"5 Exit the Application\n";
        cout<<"______________________________________________________\n";
        cout<<"Enter your option\n";
        cin>>iOption;
        cout<<"______________________________________________________\n";

        if(iOption == 1)
        {
            qobj.Display();
        }
        else if(iOption == 2)
        {
            iRetCountNode = qobj.CountNode() ;
            cout<<"Total nodes in Queue are : "<<iRetCountNode<<"\n";
        }
        else if(iOption == 3)
        {
            cout<<"Enter the data : \n";
            cin>>iNo ;

            qobj.EnQueue(iNo);
        }
        else if(iOption == 4)
        {
            iRetDeQueue = qobj.DeQueue();
            cout<<"Removed Element is : "<<iRetDeQueue<<"\n";
        }
        else if(iOption == 5)
        {
            cout<<"Thankyou for using your module\n";
            break;
        }
        else
        {
            cout<<"!! Invalid Option !!\n";
        }   
    }
    return 0 ;
}